Source/WebCore/page/ImageAnalysisQueue.cpp

```diff
@@ -158,10 +164,26 @@ void ImageAnalysisQueue::resumeProcessing()
                 m_pendingRequestCount--;
 
             resumeProcessingSoon();
+
+            if (m_queue.isEmpty() && m_imageQueueEmptyHysteresis)
+                m_imageQueueEmptyHysteresis->impulse();
         });
     }
 }
 
+void ImageAnalysisQueue::setDidBecomeEmptyCallback(Function<void()>&& callback)
+{
+    m_imageQueueEmptyHysteresis = makeUnique<PAL::HysteresisActivity>([callback = WTFMove(callback)] (PAL::HysteresisState state) {
+        if (state == PAL::HysteresisState::Stopped)
+            callback();
+    }, 1_s);
+}
+
+void ImageAnalysisQueue::clearDidBecomeEmptyCallback()
+{
+    m_imageQueueEmptyHysteresis = nullptr;
+}
+
 void ImageAnalysisQueue::clear()
 {
     // FIXME: This should cancel pending requests in addition to emptying the task queue.
```

Comment:
Let's reset `m_analysisOfAllImagesOnPageHasStarted` to `false` and `m_imageQueueEmptyHysteresis` to `nullptr` in `ImageAnalysisQueue::clear()`, as well.